src/java/main/org/apache/zookeeper/server/PurgeTxnLog.java

```diff
@@ -133,11 +133,17 @@ public boolean accept(File f){
             }
         }
         // add all non-excluded log files
-        List<File> files = new ArrayList<File>(Arrays.asList(txnLog
-                .getDataDir().listFiles(new MyFileFilter(PREFIX_LOG))));
+        List<File> files = new ArrayList<File>();
+        File[] fileArray;
+        if ((fileArray = txnLog.getDataDir().listFiles(new MyFileFilter(PREFIX_LOG))) != null) {
+            files.addAll(Arrays.asList(fileArray));
+        }
+
         // add all non-excluded snapshot files to the deletion list
-        files.addAll(Arrays.asList(txnLog.getSnapDir().listFiles(
-                new MyFileFilter(PREFIX_SNAPSHOT))));
+        if ((fileArray = txnLog.getSnapDir().listFiles(new MyFileFilter(PREFIX_SNAPSHOT))) != null) {
```

Comment:
Same as above, Its good to keep the var assignment outside if check for better readability.
`(fileArray = txnLog.getSnapDir().listFiles(new MyFileFilter(PREFIX_SNAPSHOT))`